Source/Urho3D/AngelScript/ScriptFile.cpp

```diff
@@ -321,6 +335,78 @@ bool ScriptFile::Execute(asIScriptFunction* function, const VariantVector& param
 
     scriptSystem->IncScriptNestingLevel();
     bool success = context->Execute() >= 0;

```

Comment:
Based on the Angelscript documentation, the script function returns value is only valid when the execution is actually finished (asEXECUTION_FINISHED).

```suggestion
     bool success = context->Execute() == asEXECUTION_FINISHED;
```

Reply:
Ok fixed